src/Client.php

```diff
@@ -619,7 +618,7 @@ protected function connect()
 
     public function __destruct()
     {
-        if ($this->socket) {
+        if (!is_null($this->socket)) {
```

Comment:
But I am changing it to `@fclose` and checking for `is_resource` too, just to be safe.